### DISCUSSION TITLE
MET V10.0 Series-Analysis

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#1337] [Configuration] [answered]
I'm getting warnings as follows when running Series-Analysis to generate output for multiple variables (fields) each having its own set of thresholds. In the config file I specified multiple fields in the fcst and obs dictionaries with the associated cat_thresh settings for each containing an array of thresholds unique to each variable just like I do when running Grid-Stat. The warnings are as follows:
WARNING: SeriesAnalysisConfInfo::process_config() -> "cat_thresh" should be specified in the top level of the fcst dictionary, not within each fcst.field array entry!

My input data are fcst and obs files with multiple valid times hourly over a four day period. 

It appears that one cannot run Series-Analysis for more than one field at a time if the cat_thresh cannot be set separately in each field. Is this the case and is there a work around to enable the output for all desired variables to be generated with one run. Thanks
[SeriesAnalysisConfig_m3o3_ALLVRB_fcst_obs.txt](https://github.com/dtcenter/METplus/files/7817775/SeriesAnalysisConfig_m3o3_ALLVRB_fcst_obs.txt)
.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@jwraby what you guessed is indeed the case. Series-Analysis is designed to run over one series for each run. Here's a note about that in the [Series-Analysis chapter](https://met.readthedocs.io/en/latest/Users_Guide/series-analysis.html) of the user's guide:
"This tool computes statistics for exactly one series each time it is run."

There is some flexibility in how that series is defined. You can either read the same field from multiple input files, or you can read multiple fields from the same input file. The former is how it is typically run and usually defines a time series. The latter is less often run but would enable you to evaluate multiple levels from the same input file (i.e. many pressure levels for temperature).

The biggest consideration here is memory. Many stats computed by MET require that all the input data be held in memory at the same time. For a dense grid and/or a long series, that can be a considerable amount of memory. The "block_size" config option is set conservatively VERY low (1024 by default). That's the number of grid point to process concurrently. I'd recommend increasing that as large as you can (i.e. try using the size of the domain (Nx * Ny)), to get it to run more quickly.

But I do realize that looping over many runs of Series-Analysis to process multiple variables/levels can be cumbersome. This is one place where switching to the METplus python wrappers may make your life easier. In a single METplus config file, you should be able to list all the variables/levels of interest and have METplus loop over them for you.

Hope that helps clarify.

**Comment by jwraby** [comment]
Hi John – 

Thanks for the background info which explains why in answer to my question. Also appreciate the info on alternative ways Series-Analysis can be run. I haven’t installed METPlus python wrappers, but it is intesting to learn the advantage it provides in this situation. 

R/
John

**Reply by JohnHalleyGotway** [reply]
@jwraby, great, I'll go ahead and mark this discussion as being answered and lock it to prevent future posts.

In the future, I'd recommend clicking on the "view it on GitHub" link at the bottom of the email notification for your GitHub Discussion. Then respond directly via GitHub instead of just replying to the email. When you reply to the email, your email signature (including your phone number) is included in the public post, along with all the email history. I edited your last post to remove this extra info. But responding directly via GitHub would avoid it in the future.

Thanks!
